app/controllers/lottery_controller.rb

```diff
@@ -44,20 +35,45 @@ def run_intelligent_assignment
 
   private
 
+  def construct_priority_info(sign_up_topics, teams)
+    priority_info = []
+    # Exclude any teams already signed up
+    teams_not_signing_up = teams.select{|team| SignedUpTeam.where(team_id: team.id, is_waitlisted: 0).any?}
```

Comment:
Won't this include signed up teams rather than skip them?

Reply:
You are right. Sorry for mishandling this part. Should change select to reject I think?